endpoints/openrtb2/amp_auction.go

```diff
@@ -219,6 +220,7 @@ func (deps *endpointDeps) AmpAuction(w http.ResponseWriter, r *http.Request, _ h
 		ao.Errors = append(ao.Errors, err)
 		return
 	}
+	labels.DebugEnabled = debugLog.DebugFlag
```

Comment:
I consider it a "code smell" that `debugLog` is instantiated here just to return the DebugFlag to the outer scope. It would be preferable IMHO to either calculate the debug flag prior to calling `HoldAuction` or include it as a return argument somehow. I know you didn't create this situation, but I believe it's worth an investigation to try and make this flow a bit nicer now.

Reply:
I agree with Scott here. This definitely seems odd to instantiate the `debugLog` struct here just to return the `DebugFlag` to the outer scope. How about making `debugLog` a part of the `AuctionRequest` struct? That way you can read and modify it in `HoldAuction` as needed and you can again access it outside of `HoldAuction` for setting the metrics label and such

Reply:
> That way you can read and modify it in HoldAuction as needed and you can again access it outside of HoldAuction for setting the metrics label and such

A step in the right direction, but modifying it as a way to pass information upstream is still an undesirable design. We might have to tackle that later on.